webpack.config.js

```diff
@@ -0,0 +1,84 @@
+const { exec } = require('child_process')
+const path = require('path')
+const { EsbuildPlugin } = require('esbuild-loader')
+
+const webpack = require('webpack')
+
+// Run `yarn start.plugins` to enable debug mode.
+// This mode will serve the plugin bundle via a local dev-server.
+const debug = process.env.WEBPACK_SERVE
+
+// Try exposing our socket to adb (errors are fine):
+if (process.env.WEBPACK_SERVE) {
+  console.log('adb reverse tcp:8101 tcp:8101')
+  exec('adb reverse tcp:8101 tcp:8101', () => {})
+}
+
+const bundlePath = path.resolve(
+  __dirname,
+  'android/src/main/assets/edge-currency-plugins'
+)
+
+module.exports = {
+  devtool: debug ? 'source-map' : undefined,
+  devServer: {
+    allowedHosts: 'all',
+    hot: false,
+    port: 8101,
+    static: bundlePath
+  },
+  entry: './src/index.ts',
+  mode: debug ? 'development' : 'production',
+  module: {
+    rules: [
+      {
+        exclude: /(@babel\/runtime|babel-runtime)/,
+        test: /\.ts$/,
+        use: {
+          loader: 'esbuild-loader',
+          options: { loader: 'ts' }
+        }
+      }
+    ]
+  },
+  optimization: {
+    minimizer: [
+      new EsbuildPlugin({
+        target: 'chrome67'
+      })
+    ]
+  },
+  output: {
+    chunkFilename: '[name].chunk.js',
+    filename: 'edge-currency-plugins.js',
+    path: bundlePath
+  },
+  performance: { hints: false },
+  plugins: [
+    new webpack.IgnorePlugin({ resourceRegExp: /^(https-proxy-agent)$/ }),
+    new webpack.ProvidePlugin({
+      Buffer: ['buffer', 'Buffer']
+    }),
+    new webpack.ProvidePlugin({
+      process: path.resolve('node_modules/process/browser.js')
+    })
+  ],
+  resolve: {
+    aliasFields: ['browser'],
+    extensions: ['.ts', '.js'],
+    fallback: {
+      assert: require.resolve('assert'),
+      crypto: require.resolve('crypto-browserify'),
+      fs: false,
+      http: require.resolve('stream-http'),
+      https: require.resolve('https-browserify'),
+      os: require.resolve('os-browserify/browser'),
+      path: require.resolve('path-browserify'),
+      stream: require.resolve('stream-browserify'),
+      string_decoder: require.resolve('string_decoder'),
+      url: require.resolve('url'),
+      vm: require.resolve('vm-browserify')
+    },
+    mainFields: ['browser', 'module', 'main']
+  }
```

Comment:
Now that this config file only needs to worry about edge-currency-plugins and not anybody else, I'm guessing we can really minimize this section. Does edge-currency-plugins really depend on all this Node.js stuff? Probably not all of it.

However, I understand the motivation to keep things stable for now. Maybe we can create a follow-on task and handle this cleanup in a future release.